build/common.ps1

```diff
@@ -269,22 +270,82 @@ Function Install-DotnetCLIToILMergePack {
     & $cli.DotNetExe --info
 }
 
+Function Get-LatestVisualStudioRoot {
+    param(
+        [int]$MaxVersion
+    )
+    # First try to use vswhere to find the latest version of Visual Studio.
+    if (Test-Path $VsWhereExe) {
+        if ($MaxVersion) {
+            $additionalArgs = '-version', "[15.0,$($MaxVersion+1).0)"
```

Comment:
didn't realize it, for now this will always return the VS 15.0 path and will never return VS 2019.

Reply:
Only if `MaxVersion` was passed, which it isn't in the mainline case. There is more work required to support VS2019 since "15.0" becomes "Current" going forward. This change was focusing on getting away from unsupported reads directly from our state.json file(s).

Reply:
`MaxVersion` is always passed when called from `Get-MSBuildRoot`. So my point is this is still always going to use msbuild from VS 15 even though we try to build from `VS 2019 developer command prompt`.

Reply:
Are you asking that this change to use the supported vswhere instead of using state.json directly and - in other locations - assuming the path to MSBuild is the default location, handle support for VS2019?

Reply:
I'm saying we can skip `$additionalArgs` instead always get the latest VS root and consume a corresponding MSBuild version. We can get rid of `MaxVersion` here.

Reply:
I had something like that originally, but based on how the code was, I wasn't sure if there were times you actually required a particular version with which to build: mainly, one of the cmdlets only accepted `14` or `15`. Let me pop by since you're just down the hall.

Reply:
@heaths will you have some time to update it? otherwise I can pick it and update.